back/src/adapters/secondary/pg/PgEstablishmentAggregateRepository.ts

```diff
@@ -868,30 +841,66 @@ const makeSelectImmersionSearchResultDtoQueryGivenSelectedOffersSubQuery = (
       SELECT 
       row_number,
       JSONB_BUILD_OBJECT(
-      'rome', io.rome_code, 
-      'siret', io.siret, 
-      'distance_m', io.distance_m, 
-      'name', e.name, 
-      'website', e.website, 
-      'additionalInformation', e.additional_information, 
-      'customizedName', e.customized_name, 
-      'fitForDisabledWorkers', e.fit_for_disabled_workers,
-      'position', JSON_BUILD_OBJECT('lon', e.lon, 'lat', e.lat), 
-      'romeLabel', io.rome_label,
-      'appellations',  io.appellations,
-      'naf', e.naf_code,
-      'nafLabel', public_naf_classes_2008.class_label,
-      'address', JSON_BUILD_OBJECT('streetNumberAndAddress', e.street_number_and_address, 
-                                    'postcode', e.post_code,
-                                    'city', e.city,
-                                    'departmentCode', e.department_code),
-      'contactMode', ic.contact_mode,
-      'contactDetails', JSON_BUILD_OBJECT('id', ic.uuid, 'firstName', ic.firstname, 'lastName', ic.lastname, 'email', ic.email, 'job', ic.job, 'phone', ic.phone ),
-      'numberOfEmployeeRange', e.number_employees 
+        'rome', io.rome_code, 
+        'siret', io.siret, 
+        'distance_m', io.distance_m, 
+        'isSearchable',e.is_searchable,
+        'name', e.name, 
+        'website', e.website, 
+        'additionalInformation', e.additional_information, 
+        'customizedName', e.customized_name, 
+        'fitForDisabledWorkers', e.fit_for_disabled_workers,
+        'position', JSON_BUILD_OBJECT('lon', e.lon, 'lat', e.lat), 
+        'romeLabel', io.rome_label,
+        'appellations',  io.appellations,
+        'naf', e.naf_code,
+        'nafLabel', public_naf_classes_2008.class_label,
+        'address', JSON_BUILD_OBJECT('streetNumberAndAddress', e.street_number_and_address, 
+                                      'postcode', e.post_code,
+                                      'city', e.city,
+                                      'departmentCode', e.department_code),
+        'contactMode', ic.contact_mode,
+        'contactDetails', JSON_BUILD_OBJECT('id', ic.uuid, 'firstName', ic.firstname, 'lastName', ic.lastname, 'email', ic.email, 'job', ic.job, 'phone', ic.phone ),
+        'numberOfEmployeeRange', e.number_employees 
       ) AS search_immersion_result
       FROM match_immersion_offer AS io 
       LEFT JOIN establishments AS e ON e.siret = io.siret  
       LEFT JOIN public_naf_classes_2008 ON (public_naf_classes_2008.class_id = REGEXP_REPLACE(naf_code,'(\\d\\d)(\\d\\d).', '\\1.\\2'))
       LEFT JOIN unique_establishments__immersion_contacts AS eic ON eic.establishment_siret = e.siret
       LEFT JOIN immersion_contacts AS ic ON ic.uuid = eic.contact_uuid
       ORDER BY row_number ASC; `;
+
+const offersEqual = (a: ImmersionOfferEntityV2, b: ImmersionOfferEntityV2) =>
+  // Only compare romeCode and appellationCode
+  a.romeCode === b.romeCode && a.appellationCode == b.appellationCode;
```

Comment:
Maintenant que les appellations sont obligatoires, on peut simplifier :

```suggestion
const offersEqual = (a: ImmersionOfferEntityV2, b: ImmersionOfferEntityV2) => a.appellationCode == b.appellationCode;
```